packages/kit/src/utils/routing.js

```diff
@@ -96,6 +96,61 @@ export function parse_route_id(id) {
 	return { pattern, params };
 }
 
+const basic_param_pattern = /\[(\[)?(?:\.\.\.)?(\w+?)(?:=(\w+))?\]\]?/;
+
+/**
+ * Parses a route ID, then resolves it to a route by replacing parameters with actual values from `entry`.
+ * @param {string} id The route id
+ * @param {Record<string, string | undefined>} entry The entry meant to populate the route. For example, if the route is `/blog/[slug]`, the entry would be `{ slug: 'hello-world' }`
+ * @example
+ * ```js
+ * route_from_entry(`/blog/[slug]/[...somethingElse]`, { slug: 'hello-world', somethingElse: 'something/else' }); // `/blog/hello-world/something/else`
+ * ```
+ */
+export function route_from_entry(id, entry) {
```

Comment:
the route is the input, not the output — we get a _path_ back from this function. maybe a name like this?

```suggestion
export function resolve_entry(id, entry) {
```